spec/README.md

```diff
@@ -0,0 +1,121 @@
+
+Multiple `reference` parameters should be supported if the requester wishes to categorize transactions on chain.
+
+## v0.3 Request Link
+
+This is a new proposal. No consensus or implementation exists.
+
+This proposal draws in part from https://en.bitcoin.it/wiki/BIP_0072, relying on HTTPS for transmitting and authenticating arbitrary transaction payloads. 
+
+There are a some significant shortcomings of the simple BIP21-based payment link scheme described above.
+
+#### 1. It only describes simple native SOL and SPL token transfers.
+
+Merchants, service providers, and apps may wish to mint NFTs or transfer reward tokens with purchases, invoke programs, pay gas for customer transactions, and enable many other use cases that may be developed with arbitrary transactions.
+
+Transactions on Solana must specify the accounts that will be included in the transaction upfront. Most useful instructions require the wallet signer address, their auxiliary token account address, etc.
+
+This requires knowing the wallet address and being able to generate PDAs from it, which cannot be known when the link is created. In short, it's missing a "connect wallet" function.
+
+#### 2. Payment requests are not authenticated.
+
+We may expect that payment links will be maliciously or accidentally misused. Without knowing who a receiving address belongs to, it's not possible to determine from the URL who is requesting the payment.
+
+A mechanism such as an HTTPS link allows the wallet to authenticate the source of the request. There may be other mechanisms we should consider. 
+
+---
+
+This proposal is to add an optional `request=<URL_ENCODED_URL>` parameter included in the link.
+
+An interactive protocol between the merchant and wallet follows:
+
+1. The merchant presents a QR code with the following payment link:
+```
+solana:<ADDRESS>?request=https%3A%2F%2Fmerchant.com%2Fsolanapay
+```
+Any of the parameters of the spec can also be included. `<ADDRESS>` could be considered optional. Perhaps an invalid address (e.g. `a`, `x`, or `_`) could be provided for compatibility. Regardless, it should not be used by the wallet if `request` is provided.
+
+2. The customer scans the QR code and opens their wallet app.
+
+3. The wallet parses link and prompts the user to make a request to `https://merchant.com/solanapay`.
+
+This is analogous to connecting a wallet to a dapp, so obtaining the user's permission is important for privacy.
+
+6. If permitted, the wallet makes a request to
+```
+https://merchant.com/solanapay?from=<WALLET_ADDRESS>&<...PARAMs>
+```
+The wallet should include any parameters from the URL provided, except for the `request` parameter.
+
+5. The merchant responds with a JSON object:
+```
+{"transaction":"<TRANSACTION>"}
+```
+The `transaction` field must be a base58-encoded [serialized transaction](https://solana-labs.github.io/solana-web3.js/classes/Transaction.html#serialize).
```

Comment:
Rationale for Base58 here is that it's included as a dependency of `@solana/web3.js` so it should be available in many contexts you need to work with `Transaction` objects. This may be a poor rationale for mobile wallets though. Base64 or another encoding could work.

Reply:
Base64 is my strong preference for any new work.  Base58 is such a performance landmine, and base64 support is ubiquitous.